src/Course/State.hs

```diff
@@ -193,8 +201,14 @@ distinct =
 --
 -- >>> isHappy 44
 -- True
+
+-- This is a bit confusing. I had to look it up for some reference and came across
+-- https://dev.to/diazomethan/comment/28dh . I'm gonna come back to this later.
```

Comment:
I think this is just asking you to generate the sequence of numbers  and applying the `firstRepeat` to check if the numbers start repeating.

<details>
  <summary>SPOILER</summary>
  <pre>

import Data.Char (intToDigit)

next :: Integer -> Integer
next = sum . map ((^2) . digitToInt) . show

generateSequence :: Integer -> [Integer]
generateSequence start = iterate next start
  </pre>
</details>

Reply:
Thanks a lot for reviewing this Jan. This is interesting. Using `show` didn't even cross my mind.